src/platform/OpenThread/GenericThreadStackManagerImpl_OpenThread.cpp

```diff
@@ -411,12 +411,14 @@ GenericThreadStackManagerImpl_OpenThread<ImplClass>::_StartThreadScan(NetworkCom
         mTemporaryRxOnWhenIdle = true;
         linkMode.mRxOnWhenIdle = true;
         otThreadSetLinkMode(mOTInst, linkMode);
+
+        // Delay Thread scanning to allow Child Update Request/Response transaction to finish.
+        DeviceLayer::SystemLayer().StartTimer(System::Clock::Milliseconds32(kThreadScanDelayMs), RequestThreadDiscovery, this);
+        ExitNow();
     }
 #endif
 
-    error = MapOpenThreadError(otThreadDiscover(mOTInst, 0,                       /* all channels */
-                                                OT_PANID_BROADCAST, false, false, /* disable PAN ID, EUI64 and Joiner filtering */
-                                                _OnNetworkScanFinished, this));
+    error = StartThreadDiscovery();
 
 exit:
     Impl()->UnlockThreadStack();
```

Comment:
Might be good to add code for restoring `RxOnWhenIdle` to cover the case that `DeviceLayer::SystemLayer().StartTimer` (or anything else) returns an error.

Reply:
Ahh.. Good point!